rfc/system/3553-config-override.md

```diff
@@ -0,0 +1,102 @@
+# Flyte Config Overrides RFC
+
+
+https://github.com/flyteorg/flyte/issues/475
+
+## Motivation
+As a user of Flyte, configuration like "resources (cpu/mem/gpu)", "catalog (enable/disable/version)", "retries", "spark config", "hive cluster" are created only at registration time. The only way to update is to re-register. This is not desirable, as there are times when the user may want to update these parameters for specific launchplans, or executions.
+
+Using with_overrides can provide node-level override for workflow, but this is still at compile time. Users need to use dynamic workflow if they want to pass the override values as inputs to the workflow. Moreover, with_overrides can not be applied to reference workflow because the workflow function body is not exposed.
+
+## Design
+
+By default, all task nodes are not overridable at runtime. However, users can call `.with_runtime_override(name: str)` on task nodes to register a "hook" on tasks, which allows the task to be indentified by `name` and overridden with new config at runtime.
+
+```python
+@task
+def t1():
+  ...
+
+@task
+def t2():
+  ...
+
+@workflow
+def wf():
+  t1() # this task node cannot be overridden
+  t1().with_runtime_override("task-yee") # can be overriden under the name "task-yee"
+  t2().with_runtime_override("task-ketan") # can be overriden under the name "task-ketan"
+  t3() # this task node cannot be overridden
+```
```

Comment:
```suggestion
```

One main motivation for introducing such a "hook mechanism" based on identifiers/names is that we couldn't come up with a good answer how a good UX can be achieved when having to specify overrides *in a nested workflow graph* in the UI, in code, or via the CLI. This way, the overrides can be specified in a simple map and the structure of the workflow graph does not matter/does not have to be shown.
```